pkg/gator/test/test_test.go

```diff
@@ -183,19 +182,30 @@ func TestTest(t *testing.T) {
 					t.Errorf("got err %q, want %q", err, tc.err)
 				}
 			} else if err != nil {
-				t.Errorf("got err '%v', want nil", err)
+				require.NoError(t, err)
 			}
 
 			got := resps.Results()
 
-			diff := cmp.Diff(tc.want, got, cmpopts.IgnoreFields(GatorResult{}, "Metadata", "EvaluationMeta", "EnforcementAction", "ViolatingObject"))
-			if diff != "" {
-				t.Errorf("diff in GatorResult objects (-want +got):\n%s", diff)
-			}
+			compareGatorResults(t, tc.want, got)
 		})
 	}
 }
 
+func compareGatorResults(t *testing.T, want, got []*GatorResult) {
+	t.Helper()
+
+	require.Equal(t, len(want), len(got))
+	for index, wantResult := range want {
+		gotResult := got[index]
+
+		require.Equal(t, wantResult.Constraint, gotResult.Constraint)
```

Comment:
To avoid accidental under-comparisons, we should define the relevant paths in the test case, not the execution code.

https://pkg.go.dev/github.com/google/go-cmp/cmp#FilterPath should help with doing that.

Reply:
I actually went ahead and wrote a custom option for us that slots in to the `cmpopts.IgnoreFields` call instead; let me know if you'd like me to explicitly add it to the test case struct like:

```
tcs := []struct {
		name   string
		inputs []string
		want   []*GatorResult
		err    error
		filterPath cmp.Option
	}
...
```

Reply:
Yep, in the test case.

Reply:
Added it!